compiler/optimizer/OrderBlocks.cpp

```diff
@@ -2198,7 +2198,7 @@ void checkOrderingConsistency(TR::Compilation *comp)
          {
          if (seenColdBlock)
             {
-            char *fmt= "Non-cold block_%d found after a cold block in method %s\n";
+            const char *fmt = "Non-cold block_%d found after a cold block in method %s\n";
```

Comment:
I tripped across an error in a z/OS build of OpenJ9 with this change along with eclipse-openj9/openj9#18465 for the call to `sprintf` that appears two lines down:

```
"/j9tmp/bld_62224/bld_zos_390/compiler/../omr/compiler/optimizer/OrderBlocks.cpp", line 2203.29: CCN5256 (S) A parameter of type "char *" cannot be initialized with an expression of type "const char *".
"/j9tmp/bld_62224/bld_zos_390/compiler/../omr/compiler/optimizer/OrderBlocks.cpp", line 2203.29: CCN6205 (I) The error occurred while converting to parameter 2 of "atoe_sprintf(char *, char *, ...)".
CCN0793(I) Compilation failed for file /j9tmp/bld_62224/bld_zos_390/compiler/../omr/compiler/optimizer/OrderBlocks.cpp.  Object file not created.
/j9tmp/bld_62224/bld_zos_390/compiler/../compiler/build/rules/zos-xlc/common.mk:95: recipe for target '/j9tmp/bld_62224/bld_zos_390/compiler/../objs/omr/compiler/optimizer/OrderBlocks.o' failed
make[3]: *** [/j9tmp/bld_62224/bld_zos_390/compiler/../objs/omr/compiler/optimizer/OrderBlocks.o] Error 12
```

To avoid extra delays in getting your changes merged, I would suggest you add a `const_cast<char *>(fmt)` to the call to sprintf, and mention this in the issue you had opened for additional const-ness clean up.

I believe that `sprintf` is defined in terms of `atoe_sprintf` on z/OS in [`util/a2e/headers/stdio.h`](https://github.com/eclipse/omr/blob/0d4b2da9e86466d10e8f9a77cc3521c4b9548341/util/a2e/headers/stdio.h#L62) and [`util/a2e/atoe.c`](https://github.com/eclipse/omr/blob/0d4b2da9e86466d10e8f9a77cc3521c4b9548341/util/a2e/atoe.c#L1344).  You should mention them in that issue as well.